components/camel-pulsar/src/main/java/org.apache.camel.component.pulsar/PulsarComponent.java

```diff
@@ -1,24 +1,69 @@
 package org.apache.camel.component.pulsar;
 
+import java.util.Collections;
+import java.util.List;
 import java.util.Map;
+
 import org.apache.camel.CamelContext;
 import org.apache.camel.Endpoint;
+import org.apache.camel.component.pulsar.configuration.AdminConfiguration;
 import org.apache.camel.component.pulsar.configuration.PulsarEndpointConfiguration;
 import org.apache.camel.impl.DefaultComponent;
+import org.apache.pulsar.client.admin.PulsarAdmin;
+import org.apache.pulsar.client.admin.PulsarAdminException;
+import org.apache.pulsar.client.api.PulsarClientException;
+import org.apache.pulsar.client.impl.conf.ClientConfigurationData;
+import org.apache.pulsar.common.policies.data.TenantInfo;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
 
 public class PulsarComponent extends DefaultComponent {
 
-    public PulsarComponent(CamelContext context) {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PulsarComponent.class);
+
+    private final PulsarEndpointConfiguration configuration;
+    private final AdminConfiguration adminConfiguration;
+
+    PulsarComponent(CamelContext context, PulsarEndpointConfiguration configuration, AdminConfiguration adminConfiguration) {
```

Comment:
I assumed this constructor is called internally by the camel context?

Reply:
Going to add properties and default constructor to allow both ways of configuring. Spring is moving to constructor only injection and deprecating property autowiring.

Reply:
Done